crates/weaver_semconv/src/any_value.rs

```diff
@@ -0,0 +1,690 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(rustdoc::invalid_html_tags)]
+
+//! AnyValue specification.
+
+use std::fmt::{Display, Formatter};
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::attribute::{BasicRequirementLevelSpec, EnumEntriesSpec, Examples, RequirementLevel};
+use crate::stability::Stability;
+
+/// The AnyValueTypeSpec is a specification of a value that can be of any type.
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Hash, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+#[serde(tag = "type")]
+pub enum AnyValueSpec {
+    /// A boolean attribute.
+    Boolean {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// A integer attribute (signed 64 bit integer).
+    Int {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// A double attribute (double precision floating point (IEEE 754-1985)).
+    Double {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// A string attribute.
+    String {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// An array of strings attribute.
+    #[serde(rename = "string[]")]
+    Strings {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// An array of integer attribute.
+    #[serde(rename = "int[]")]
+    Ints {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// An array of double attribute.
+    #[serde(rename = "double[]")]
+    Doubles {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// An array of boolean attribute.
+    #[serde(rename = "boolean[]")]
+    Booleans {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// The value type is a map of key, value pairs
+    Map {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+        /// The collection of key, values where the value is an `AnyValueSpec`
+        fields: Vec<AnyValueSpec>,
+    },
+
+    /// The value type will just be a bytes.
+    Bytes {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec
+    },
+
+    /// The value type is not specified.
+    Undefined {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec
+    },
+
+    /// An enum definition type.
+    Enum {
```

Comment:
Add (even though we talked about) see additional comments in the `event.yaml` in the `data` folder.